File: VB/Q248711/Form1.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports DevExpress.XtraEditors
Imports DevExpress.XtraPivotGrid
Imports System.Collections.Generic
Imports DevExpress.XtraGrid.Views.Base
Imports DevExpress.XtraGrid.Views.Grid

Namespace Q248711
	Partial Public Class Form1
		Inherits XtraForm

		Public Sub New()
			InitializeComponent()
			AddHandler gridView1.FocusedRowChanged, AddressOf OnGridViewFocusedRowChanged
			AddHandler gridView1.FocusedColumnChanged, AddressOf OnGridViewFocusedColumnChanged
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			' TODO: This line of code loads data into the 'nwindDataSet.SalesPerson' table. You can move, or remove it, as needed.
			Me.salesPersonTableAdapter.Fill(Me.nwindDataSet.SalesPerson)

		End Sub

		Private Sub OnGridViewFocusedColumnChanged(ByVal sender As Object, ByVal e As FocusedColumnChangedEventArgs)
			UpdatePivotSelection(pivotGrid, e.FocusedColumn.FieldName, DirectCast(sender, GridView).GetFocusedValue())
		End Sub

		Private Shared Sub UpdatePivotSelection(ByVal pivot As PivotGridControl, ByVal fieldName As String, ByVal value As Object)
			Dim field As PivotGridField = pivot.Fields(fieldName)
			If field.Area = PivotArea.DataArea OrElse field.Area = PivotArea.FilterArea Then
				pivot.Cells.Selection = Rectangle.Empty
			End If
			Dim count As Integer = If(field.Area = PivotArea.ColumnArea, pivot.Cells.ColumnCount, pivot.Cells.RowCount)
			Dim range As New List(Of Integer)()
			For i As Integer = 0 To count - 1
				If Comparer.Default.Compare(value, pivot.GetFieldValue(field, i)) = 0 Then
					range.Add(i)
				End If
			Next i
			If range.Count = 0 Then
				Return
			End If
			pivot.Cells.Selection = If(field.Area = PivotArea.ColumnArea, New Rectangle(range(0), 0, range.Count, pivot.Cells.RowCount), New Rectangle(0, range(0), pivot.Cells.ColumnCount, range.Count))
			DirectCast(pivot, IPivotGridViewInfoDataOwner).DataViewInfo.ViewInfo.LeftTopCoord = pivot.Cells.Selection.Location
		End Sub

		Private Sub OnGridViewFocusedRowChanged(ByVal sender As Object, ByVal e As FocusedRowChangedEventArgs)
			Dim view As GridView = DirectCast(sender, GridView)
			UpdatePivotSelection(pivotGrid, view.FocusedColumn.FieldName, view.GetFocusedValue())
		End Sub
	End Class
End Namespace

File: CS/Q248711/Form1.cs
using System;
using System.Drawing;
using System.Collections;
using DevExpress.XtraEditors;
using DevExpress.XtraPivotGrid;
using System.Collections.Generic;
using DevExpress.XtraGrid.Views.Base;
using DevExpress.XtraGrid.Views.Grid;

namespace Q248711 {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
            gridView1.FocusedRowChanged += OnGridViewFocusedRowChanged;
            gridView1.FocusedColumnChanged += OnGridViewFocusedColumnChanged;
        }

        private void Form1_Load(object sender, EventArgs e) {
            // TODO: This line of code loads data into the 'nwindDataSet.SalesPerson' table. You can move, or remove it, as needed.
            this.salesPersonTableAdapter.Fill(this.nwindDataSet.SalesPerson);

        }

        private void OnGridViewFocusedColumnChanged(object sender, FocusedColumnChangedEventArgs e) {
            UpdatePivotSelection(pivotGrid, e.FocusedColumn.FieldName, ((GridView)sender).GetFocusedValue());
        }

        private static void UpdatePivotSelection(PivotGridControl pivot, string fieldName, object value) {
            PivotGridField field = pivot.Fields[fieldName];
            if (field.Area == PivotArea.DataArea || field.Area == PivotArea.FilterArea) 
                pivot.Cells.Selection = Rectangle.Empty;
            int count = field.Area == PivotArea.ColumnArea ? pivot.Cells.ColumnCount : pivot.Cells.RowCount;
            List<int> range = new List<int>();
            for (int i = 0; i < count; i++)
                if (Comparer.Default.Compare(value, pivot.GetFieldValue(field, i)) == 0) range.Add(i);
            if (range.Count == 0) return;
            pivot.Cells.Selection = field.Area == PivotArea.ColumnArea ? 
                new Rectangle(range[0], 0, range.Count, pivot.Cells.RowCount) :
                new Rectangle(0, range[0], pivot.Cells.ColumnCount, range.Count);
            ((IPivotGridViewInfoDataOwner)pivot).DataViewInfo.ViewInfo.LeftTopCoord =
                pivot.Cells.Selection.Location;
        }

        private void OnGridViewFocusedRowChanged(object sender, FocusedRowChangedEventArgs e) {
            GridView view = (GridView)sender;
            UpdatePivotSelection(pivotGrid, view.FocusedColumn.FieldName, view.GetFocusedValue());
        }
    }
}